script.js

```diff
@@ -0,0 +1,224 @@
+                { text: document.getElementById('optionD').value, isCorrect: document.querySelector('input[name="correct"][value="4"]').checked }
+            ];
+
+            // Create a new question object and add it to the quizQuestions array
+            quizQuestions.push({ question, options });
+
+            // Update the question list display
+            updateQuestionsList();
+
+            // Reset the form inputs for new entries
+            event.target.reset();
+        });
+    }
+
+    // Search functionality for filtering questions
+    const searchButton = document.getElementById('search-button');
+    if (searchButton) {
+        searchButton.addEventListener('click', () => {
+            const searchTerm = document.getElementById('search-input').value.toLowerCase();
+            
+            // Filter questions based on the search term
+            const filteredQuestions = quizQuestions.filter(q => q.question.toLowerCase().includes(searchTerm));
+            
+            // Display the filtered questions
+            displayFilteredQuestions(filteredQuestions);
+        });
+    }
+
+    // Display filtered questions dynamically
+    const displayFilteredQuestions = (filteredQuestions) => {
+        const questionsList = document.getElementById('questions-list');
+        questionsList.innerHTML = ''; // Clear the existing list
+
+        filteredQuestions.forEach((quizQuestion, index) => {
+            const questionDiv = document.createElement('div');
+            questionDiv.classList.add('question-item');
+
+            // Display the question text
+            const questionText = document.createElement('p');
+            questionText.textContent = `${index + 1}. ${quizQuestion.question}`;
+            questionDiv.appendChild(questionText);
+
+            // Display options as a list
+            const optionsList = document.createElement('ul');
+            quizQuestion.options.forEach(option => {
+                const optionItem = document.createElement('li');
+                optionItem.textContent = option.text;
+                optionsList.appendChild(optionItem);
+            });
+            questionDiv.appendChild(optionsList);
+
+            // Reveal correct answer button
+            const revealButton = document.createElement('button');
+            revealButton.textContent = 'Reveal Correct Answer';
+            revealButton.addEventListener('click', () => {
+                const correctOptions = quizQuestion.options
+                    .filter(option => option.isCorrect)
+                    .map(option => option.text)
+                    .join(', ');
+                alert(`Correct Answer(s): ${correctOptions}`);
+            });
+            questionDiv.appendChild(revealButton);
+
+            questionsList.appendChild(questionDiv);
+        });
+    };
+
+    // Sorting functionality (alphabetical or random)
+    const sortOptions = document.getElementById('sort-options');
```

Comment:
I see that you have implemented the sorting of the questions, but you have no element in the DOM which has the ID `#sort-options`. Therefore this functionality cannot be used.